assets/js/modules/analytics/datastore/accounts.js

```diff
@@ -396,7 +305,18 @@ export const selectors = {
 	 * @return {boolean} Whether accounts are currently being fetched or not.
 	 */
 	isDoingGetAccounts( state ) {
-		return !! state.isFetchingAccountsPropertiesProfiles;
+		// Check if dispatch calls right after fetching are still awaiting.
+		if ( state.isAwaitingAccountsPropertiesProfilesCompletion ) {
+			return true;
+		}
+		// Since isFetchingGetAccountsPropertiesProfiles (via createFetchStore)
+		// holds information based on specific values but we only need
+		// generic information here, we need to check whether ANY such
+		// request is in progress.
+		if ( 'object' !== typeof state.isFetchingGetAccountsPropertiesProfiles ) {
+			return false;
+		}
+		return Object.values( state.isFetchingGetAccountsPropertiesProfiles ).some( ( value ) => value );
```

Comment:
Would need to become a registry selector, but now about this?
```suggestion
		return select( STORE_NAME ).isResolving( 'getAccounts' );
```

Reply:
I generally like this suggestion, however since this would couple the function to the resolver, at that point we might as well have components that need this information call `isResolving` themselves. I wonder whether we should rather rely on that than introducing an unnecessary selector.

Reply:
Yeah, I don't think the logic can be entirely replaced by the one selector, but I think this should cover it:
```js
return select( STORE_NAME ).isResolving( 'getAccounts' )
    || return select( STORE_NAME ).isFetchingGetAccountsPropertiesProfiles()
```
In most cases the resolver is what makes the request (I believe tests are the only current exception here), so there is already some coupling to the resolver in that way. However, it is possible to dispatch the action directly so that's what the "OR" logic handles. As an aside, I don't think the `typeof isFetchingGetAccountsPropertiesProfiles` checks are necessary anymore now that `INITIAL_STATE[ isFetching ]` is set.

I would much more prefer using something built-in that we already have and omitting a `reducerCallback` for these "combined requests" than adding new state and actions as a workaround.

I'm not suggesting we make this change everywhere, that's indeed a conversation for another issue, but I think it makes sense for `isDoingGetAccounts` and `isDoingGetProperties`.

Reply:
I'd rather revisit this separately throughout, potentially our `isDoingGet*` selectors are entirely unnecessary and we should instead rely on checking resolvers only. This would also mean though that, alongside `receive*` calls in test setups we would also need to mark the relevant resolvers as finished.